Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/PANOSPolicyOptimizer_test.py

```diff
@@ -0,0 +1,120 @@
+import pytest
+from CommonServerPython import *
+from PANOSPolicyOptimizer import Client, policy_optimizer_get_rules_command
+
+
+BASE_URL = 'https://test.com'
+
+
+def get_firewall_instance_client():
+    return Client(url=BASE_URL, username='test', password='test', vsys='test', device_group='', verify=False, tid=0)
+
+
+def get_panorama_instance_client():
+    return Client(url=BASE_URL, username='test', password='test', vsys='', device_group='test', verify=False, tid=0)
+
+
+QUERYING_RULES_PARAMS = [
+    (
+        get_firewall_instance_client(),
+        'main',
+    ),
+    (
+        get_panorama_instance_client(),
+        'pre',
+    ),
+    (
+        get_panorama_instance_client(),
+        'post',
+    )
+]
+
+
+@pytest.mark.parametrize("client, position", QUERYING_RULES_PARAMS)
+def test_body_request_is_valid_when_querying_rules(mocker, client, position):
+    """
+    Given
+        - a client.
+
+    When
+        - querying rules in firewall/panorama instances.
+
+    Then
+        - Verify that the body request that was sent is correct for each type of instance.
+    """
+    mocker.patch.object(client, 'token_generator', return_value='123')
+    response = requests.Response()
+    response._content = b'{"type":"rpc","tid":"51","action":"PanDirect","method":"run",' \
```

Comment:
Can we move all mocked content responses and expected results to dedicated files? it will make the tests much more readable.

Reply:
yes, you are right, ill set it up in a file